File: Python3/1310. XOR Queries of a Subarray.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        xor = [0]
        for i in arr:
            xor.append(xor[-1] ^ i)

        res = []

        for start, end in queries:
            res.append(xor[end + 1] ^ xor[start])

        return res 

File: C++/1310. XOR Queries of a Subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        vector<int> xorArr = {0};
        for(auto x:arr) {
            xorArr.push_back(xorArr.back() ^ x);
        }
        vector<int> res;
        for(auto iter:queries) {
            res.push_back(xorArr[iter[1] + 1] ^ xorArr[iter[0]]);
        }
        return res;
    }
};
